src/Raven.Server/Documents/Versioning/VersioningStorage.cs

```diff
@@ -107,6 +118,22 @@ public void InitializeFromDatabaseRecord(DatabaseRecord dbRecord)
                     return;
 
                 Configuration = dbRecord.Versioning;
+
+                using (var tx = _database.DocumentsStorage.Environment.WriteTransaction())
```

Comment:
I don't like the direct use of the write transaction here.
Can't we do that as part of the actual versioning ? IIRC, we already call that as part of using versioning, no?
Why do we need that specially?